### ISSUE TITLE
What is the "core": entry used for in views .json config files?

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#2432]
Now that I am creating PRs that involve creating default views configs, I noticed that there are these lines added in the config files. What are they used for? Should we remove them if not used? Should we also remove the function that adds them in the first place?

**Comment by BWPanda** [comment]
I'm guessing it comes from Drupal. The Drupal `views_views` tables says this about the `core` field:

> Stores the drupal core version of the view.

So no great revelations there...

**Comment by klonos** [comment]
...between 2016 and last October, someone noticed it in D8 as well 😅 https://www.drupal.org/project/drupal/issues/3087692

Change record: https://www.drupal.org/node/3087832

> In Drupal 8 views configuration no longer stores the version of Drupal the view has been created with. If your module, theme, or profile has exported views in its config/install or config/optional directories then this will be need to be removed.
...
> This key will be removed from active configuration on update and even if a module does have exported configuration with the key it will be removed on module install.

**Comment by klonos** [comment]
...the PR accounts for existing views, by removing the `"core"` key from config via an update hook. They key is not being removed from imported views, because the import process uses a generic config import - but if imported views are either cloned, or edited and saved, the key is removed then.

**Comment by BWPanda** [comment]
@klonos I don't mean to disparage your hard work, but it's really hard doing a code review when there are lots of unrelated changes (e.g. all the comment updates). Can these be moved to a separate issue?

**Comment by klonos** [comment]
@BWPanda you are absolutely right (although I did add a self-CR comment, to point the single non-comment change in the `view.inc` file) ...I've brought this up during the dev meeting, and it is preferable to split the comment-related  changes out into their own issue/PR.

I will fix...

**Comment by BWPanda** [comment]
> although I did add a self-CR comment, to point the single non-comment change in the `view.inc` file

I noticed that, but it seems even you missed https://github.com/backdrop/backdrop/pull/3090/files#diff-9ef413d342527d917f7cfd8b7cd7d9ebL1806 :wink:

**Comment by klonos** [comment]
touché 😅

**Comment by klonos** [comment]
...comment changes moved to #4333

I've updated the PR here, to revert any comment-related changes; so back to NR 😉

**Comment by BWPanda** [comment]
Left a review requesting changes (well, one small change and one possible change depending on your opinion).

Also, I don't think this is a bug. It has no ill effects on the site/user, doesn't cause problems, etc.

**Comment by klonos** [comment]
I've addressed your comments in the PR @BWPanda 👍 ...and yes, this is not a bug per se; just an annoyance. I have categorized this issue as a task, and tasks are allowed to be committed at any time (so bug releases too) - only things that are deemed new features/functionality need to wait till next minor.

**Comment by bd0bd** [comment]
Well, I do not use Backdrop CMS any more and I am not going to use it in future.
Drupal 7 is better suitable for business.
Bye Backdrop!

**Comment by klonos** [comment]
Sorry to see you go @bd0bd. You've provided some interesting feedback in the issue queue over the years. Good luck with D7 ...until Nov 2021, at which point we'll either see you again I guess 😅 ...unless D8/9 becomes an option for you by then.

**Comment by indigoxela** [comment]
This branch has conflicts that must be resolved, probably just the update hook number, so an easy fix. Needs work, though.
